spring-cloud-stream-binders/spring-cloud-stream-binder-redis/src/main/java/org/springframework/cloud/stream/binder/redis/RedisMessageChannelBinder.java

```diff
@@ -256,16 +232,14 @@ public Boolean recover(RetryContext context) throws Exception {
 	}
 
 	@Override
-	public Binding<MessageChannel> bindProducer(final String name, MessageChannel moduleOutputChannel, Properties properties) {
+	protected Binding<MessageChannel> doBindProducer(final String name, MessageChannel moduleOutputChannel, ProducerProperties properties) {
 		Assert.isInstanceOf(SubscribableChannel.class, moduleOutputChannel);
-		validateProducerProperties(name, properties, SUPPORTED_PRODUCER_PROPERTIES);
-		RedisPropertiesAccessor accessor = new RedisPropertiesAccessor(properties);
-		return doRegisterProducer(name, moduleOutputChannel, accessor);
+		return doRegisterProducer(name, moduleOutputChannel, properties);
 	}
 
-	private RedisQueueOutboundChannelAdapter createProducerEndpoint(String name, RedisPropertiesAccessor accessor) {
-		String partitionKeyExtractorClass = accessor.getPartitionKeyExtractorClass();
-		Expression partitionKeyExpression = accessor.getPartitionKeyExpression();
+	private RedisQueueOutboundChannelAdapter createProducerEndpoint(String name, ProducerProperties properties) {
+		String partitionKeyExtractorClass = properties.getPartitionKeyExtractorClass();
+		String partitionKeyExpression = properties.getPartitionKeyExpression();
```

Comment:
should the `partitionKeyExpression` be an `Expression` type within the properties object? (there is a Converter available for that conversion when properties are bound... we rely on that for some of the out-of-the-box `spring-cloud-stream` apps)